enterprise_access/apps/api/v1/views/subsidy_access_policy.py

```diff
@@ -454,15 +454,19 @@ def redeem(self, request, *args, **kwargs):
             # point, we should also consider migrating this logic into the policy model so that different policy types
             # that have different locking needs can supply different lock kwargs.
             with policy.lock():
-                if policy.can_redeem(lms_user_id, content_key):
-                    redemption_result = policy.redeem(lms_user_id, content_key, metadata)
+                can_redeem, reason, existing_transactions = policy.can_redeem(lms_user_id, content_key)
+                if can_redeem:
+                    historical_redemptions = [r.uuid for r in existing_transactions if r.state == 'committed']
```

Comment:
hmm i think for the sake of idempotency, we need all historical transactions - not just successful ones.

Reply:
...or maybe have your key-generating utility function do the sorting so the caller doesn't have to think too hard?

Reply:
oh i typed 'failed' when I meant to type 'pending' in my force-push.  Fixing again.  Once I get unit tests up this should be covered too.

Reply:
I'm re-opening this thread due to developing a fairly strong opinion that "failed" and "reversed" transactions should version an idempotency key.  Please see full explanation in docstring for `_redemptions_for_idempotency_key()`.  Thoughts?